New resource: `web pubsub`

xiaxyi opened the pull request:
This PR is to support the Azure resource - Web Pubsub.

Acc test:
```
--- PASS: TestAccWebpubsub_withPropertyUpdate (743.67s)
--- PASS: TestAccWebpubsub_standard (743.80s)
--- PASS: TestAccWebpubsub_basic (455.21s)
--- PASS: TestAccWebpubsub_complete (455.58s)
--- PASS: TestAccWebpubsub_requiresImport (470.58s)
--- PASS: TestAccWebpubsub_skuUpdate (1025.53s)
--- PASS: TestAccDataSourceWebPubsub_basic (410.87s)


--- PASS: TestAccWebPubsubHub_basic (407.87s)
--- PASS: TestAccWebPubsubHub_complete (473.54s)
--- PASS: TestAccWebPubsubHub_requiresImport (490.00s)
--- PASS: TestAccWebPubsubHub_withAuthUpdate (631.87s)
--- PASS: TestAccWebPubsubHub_withPropertyUpdate (662.32s)
--- PASS: TestAccWebPubsubHub_withMultipleEventhandlerSettingsUpdate (662.34s)
--- PASS: TestAccDataSourceWebPubsubHub_basic (370.58s)
```

mybayern1974 commented:
This is the PR to address https://github.com/hashicorp/terraform-provider-azurerm/issues/14579

katbyte commented on website/docs/d/web_pubsub_hub.html.markdown:
```diff
@@ -0,0 +1,45 @@
+---
```
if we are going to create this data source could we add all the properties instead of leaving it barebones like this?

katbyte commented on .teamcity/components/generated/services.kt:
```diff
@@ -97,5 +97,6 @@ var services = mapOf(
         "videoanalyzer" to "Video Analyzer",
-        "web" to "Web"
+        "web" to "Web",
+        "webpubsub" to "Webpubsub"
```
could we use a property description here?
```suggestion
        "webpubsub" to "Web Publishing Subscription"
```

katbyte commented on website/docs/r/web_pubsub_hub.html.markdown:
```diff
@@ -0,0 +1,126 @@
+---
+subcategory: "Web Pubsub"
```
what is a "pubsub"? should this be

```suggestion
subcategory: "Web Publishing Subscription"
```
maybe?

katbyte commented on internal/services/webpubsub/webpubsub_resource.go:
```diff
@@ -0,0 +1,411 @@
+				ValidateFunc: validation.IntInSlice([]int{1, 2, 5, 10, 20, 50, 100}),
+			},
+
+			"live_trace_configuration": {
```
we don't need configuration here as it is implied?
```suggestion
			"live_trace": {
```

katbyte commented on internal/services/webpubsub/webpubsub_hub_resource.go:
```diff
@@ -0,0 +1,314 @@
+				},
+			},
+
+			"anonymous_connect_enabled": {
```
grammer?
```suggestion
			"anonymous_connections_enabled": {
```

katbyte reviewed (changes requested):
Thanks @xiaxyi - gave this a quick review and left some comments inline

xiaxyi commented:
Thanks @katbyte for the suggestions, considering the fact that the hub properties are not commonly used, I removed the data source, as for the others, I modified the code accordingly.

catriona-m reviewed (changes requested):
Thanks @xiaxyi this is looking really good. I spotted a couple of minor typos/formatting that could be changed and left some suggestions inline.

For consistency, it would also be good if all the filenames followed the pattern "web_pubsub*".

Thanks!

catriona-m commented:
It looks like there are also some test failures:

```
Test Failed
------- Stdout: -------
=== RUN   TestAccWebPubsubHub_basic
=== PAUSE TestAccWebPubsubHub_basic
=== CONT  TestAccWebPubsubHub_basic
=== CONT  TestAccWebPubsubHub_basic
testing_new.go:70: Error running post-test destroy, there may be dangling resources: exit status 1
Error: deleting "Web Pubsub: (Web Pub Sub Name \"acctest-webpubsub-220120152314425032\" / Resource Group \"acctestRG-wps-220120152314425032\")": webpubsub.Client#Delete: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> Code="Conflict" Message="There is another running operation. You cannot delete the resource before it's completed."
--- FAIL: TestAccWebPubsubHub_basic (242.52s)
```

xiaxyi commented:
Thanks @catriona-m for the suggestion. I made the change for the resource to look more consistence.

regarding the acc test, I tried multiple times from my side but not able to reproduce the same error:
```
--- PASS: TestAccWebpubsub_basic (443.01s)
--- PASS: TestAccWebpubsubHub_complete (460.65s)
--- PASS: TestAccWebpubsubHub_basic (471.04s)
--- PASS: TestAccWebpubsubHub_withAuthUpdate (684.59s)
--- PASS: TestAccWebpubsubHub_withMultipleEventhandlerSettingsUpdate (716.00s)
--- PASS: TestAccWebpubsub_withPropertyUpdate (772.24s)
--- PASS: TestAccWebpubsub_standard (797.04s)
--- PASS: TestAccWebpubsubHub_requiresImport (545.09s)
--- PASS: TestAccWebpubsub_requiresImport (530.36s)
--- PASS: TestAccWebpubsub_skuUpdate (1139.22s)
--- PASS: TestAccWebpubsubHub_withPropertyUpdate (736.97s)
--- PASS: TestAccWebpubsub_complete (523.44s)
```
I had an offline discussion with my colleague from web pubsub team, he checked the logs and said we sent another deletion request though the first deletion request was accepted. Do we have any retry logics in the script? are you able to reproduce the same error running the acc test in your environment?

catriona-m reviewed (approved):
Thanks for updating this @xiaxyi - the tests are now passing and this LGTM 🎉
